### PR TITLE
Tutorial added to Tank Operation activity

### PR DESCRIPTION
**Pull request by ksraj123**
Fixes Issue #597 
- [x] Should display a welcome message to explain the activity (could reuse the help message describing the activity in Sugarizer activities List view)
- [x] Should explain usage of each button in the toolbar
- [x] Should explain usage of each element in the UI
- [x] Should be localized
- [x] Should respect the Sugarizer UI, as explained in tutorial

- ENGLISH
tutorial

- FRENCH
tutorial-french

### CODE REVIEW

File: activities/TankOp.activity/css/activity.css [hunk 1]
```diff
@@ -18,4 +22,132 @@
   position: absolute;
   visibility: hidden;
   z-index: 5;
-} 
\ No newline at end of file
+}
+
+.tutorial-title {
+	background-color: #808080 !important;
```
Comment (llaske) [inline activities/TankOp.activity/css/activity.css]: Do not mix indentation style in the same file

File: activities/TankOp.activity/lib/settings.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ define(["webL10n","sugar-web/env"], function (webL10n,env) {
 		env.getEnvironment(function(err, environment) {
 			var defaultLanguage = (typeof chrome != 'undefined' && chrome.app && chrome.app.runtime) ? chrome.i18n.getUILanguage() : navigator.language;
 			var language = environment.user ? environment.user.language : defaultLanguage;
+			console.log(environment.user.language);
```
Comment (llaske) [inline activities/TankOp.activity/lib/settings.js]: Remove debug message

File: activities/TankOp.activity/lib/tutorial.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+				orphan: true,
+				placement: "bottom",
+				title: l10n.get("TutoExplainTitle"),
+				content: l10n.get("TutoExplainContent")
+			},
+			{
+				element: "#fullscreen-button",
+				placement: "bottom",
+				title: l10n.get("TutoFullScreenButtonTitle"),
+				content: l10n.get("TutoFullScreenButtonContent")
```
Comment (llaske) [inline activities/TankOp.activity/lib/tutorial.js]: Do not mix indentation style in the same file

File: activities/TankOp.activity/locale.ini [hunk 4]
```diff
@@ -95,6 +127,22 @@ Joshfeed=joshfeed from freesound (raté)
 Danipenet=DANIpeNet from freesound (mission terminée)
 Juskiddink=juskiddink from freesound (mission echouée)
 
+TutoPrev=Préc
+TutoNext=Suiv
+TutoEnd=Fin
+TutoExplainTitle=Activité d'exploitation de chars
```
Comment (llaske) [inline activities/TankOp.activity/locale.ini]: Don't do automatic translation, you could use the english localization for all languages

File: activities/TankOp.activity/css/activity.css (hunk 1 above)
Comment (ksraj123) [reply activities/TankOp.activity/css/activity.css]: Indention Issue Fixed in activity.css file

File: activities/TankOp.activity/lib/settings.js (hunk 2 above)
Comment (ksraj123) [reply activities/TankOp.activity/lib/settings.js]: Debug message removed.

File: activities/TankOp.activity/lib/tutorial.js (hunk 3 above)
Comment (ksraj123) [reply activities/TankOp.activity/lib/tutorial.js]: Indention made uniform across tutorial.js file

File: activities/TankOp.activity/locale.ini (hunk 4 above)
Comment (ksraj123) [reply activities/TankOp.activity/locale.ini]: @llaske Should I use english localization for all titles or should I just use it for only these 4?

Comment (s-mag) [reply activities/TankOp.activity/locale.ini]: Use it for all. Native speakers will translate them

Comment (ksraj123) [reply activities/TankOp.activity/locale.ini]: @llaske automatically translated text replaced with english.

@sashr - Thank you for clarification.
> Use it for all. Native speakers will translate them

**Comment by ksraj123** [comment]
@llaske Tutorial added to play screen.
play mode tutorials

File: activities/TankOp.activity/images/go-right.svg [hunk 5]
```diff
@@ -43,15 +43,15 @@
      inkscape:current-layer="svg2" /><g
      display="block"
      id="go-right"
-     style="fill:#333333">
+     style="fill:#ffffff">
```
Comment (llaske) [inline activities/TankOp.activity/images/go-right.svg]: You're not supposed to change this. It should be adapted automatically when disable/enable

File: activities/TankOp.activity/index.html [hunk 6]
```diff
@@ -42,6 +49,7 @@
 	  <button class="toolbutton" id="activity-button" title="Tank Operation"></button>
 	  <p class="title">TANK OPERATION</p>
 	  <button class="toolbutton pull-right" id="stop-button" title="Stop"></button>
+	  <button class="toolbutton pull-right" id="help-button" title="Tutorial"></button>
```
Comment (llaske) [inline activities/TankOp.activity/index.html]: The help button should be to the left of fullscreen button

File: activities/TankOp.activity/lib/tutorial.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
+define(["webL10n"], function (l10n) {
+	var tutorial = {};
+
+	tutorial.start = function(play) {
+		var homeSteps = [
```
Comment (llaske) [inline activities/TankOp.activity/lib/tutorial.js]: Still some indentation mix, check your editor:

image

File: activities/TankOp.activity/index.html (hunk 6 above)
Comment (ksraj123) [reply activities/TankOp.activity/index.html]: Order Corrected.

File: activities/TankOp.activity/images/go-right.svg (hunk 5 above)
Comment (ksraj123) [reply activities/TankOp.activity/images/go-right.svg]: Modified files replaced with original svg files. Working as expected now.

File: activities/TankOp.activity/lib/tutorial.js (hunk 7 above)
Comment (ksraj123) [reply activities/TankOp.activity/lib/tutorial.js]: Indention mix issue in tutorials.js is resolved.

**Comment by llaske** [comment]
Good.

Still few issues:

* The arrow to change mission are now invisible

image

I guess it's related to a conflict with existing icons. In that case, just duplicate icons to avoid the conflict.

* Miss a popup to explain missions completed zone
* I suggest to set position for popup on start button at top instead of bottom to avoid scrollbar
* I suggest you to move popup for Credits, Fullscreen and Stop at the end of the array
* I suggest to add a popup in Play mode to explain the meaning of LCD display

**Comment by ksraj123** [comment]
@llaske Modifications made as per suggestions.

- Fixed Invisible mission changing icon issue
- Added a popup for mission completed zone
- Changed the position for start button popup to top
- Moved the popup for credits, full screen and stop to the last of the array
- Added a pop up for the canvas in play mode
- Added popups for wave count and score
- Modified the popup for virtual keyboard to select the entire keyboard not just first three buttons
- Modified the popup for change mission to select the entire thing not just right button

Home Screen
homeModifier

Play Screen
playModified

**Comment by llaske** [comment]
Good job. Thanks.

**Pull request merged**